File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
void firstPos(vector<int> &nums,int target,int &first){
    int n=nums.size();
    int left=0;
    int right=n-1;
    while(left<=right){
        int mid =left+(right-left)/2;
        if(nums[mid]==target){
            first=mid;
            right=mid-1;
        }
        else if(nums[mid]>target){
            right=mid-1;

        }else{
            left=mid+1;
        }
    }
}
void lastPos(vector<int> &nums,int target,int &first){
    int n=nums.size();
    int left=0;
    int right=n-1;
    while(left<=right){
        int mid =left+(right-left)/2;
        if(nums[mid]==target){
            first=mid;
            left=mid+1;
        }
        else if(nums[mid]>target){
            right=mid-1;

        }else{
            left=mid+1;
        }
    }
}

    vector<int> searchRange(vector<int>& nums, int target) {
        int firstAns=-1;
        int lastAns=-1;
        firstPos(nums,target,firstAns);
        lastPos(nums,target,lastAns);       
        vector<int> result(2);
        result[0]=firstAns;
        result[1]=lastAns;
        return result;
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
   private int firstocc(int[] nums,int target){
        int low=0,first=-1;
        int high=nums.length-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(nums[mid]==target){
                first=mid;
                high=mid-1;
            }
            else if(nums[mid]<target){
                low=mid+1;
            }
            else high=mid-1;
        }
        return first;
    }
   private int lastocc(int[] nums,int target){
        int low=0,last=-1;
        int high=nums.length-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(nums[mid]==target){
                last=mid;
                low=mid+1;
            }
            else if(nums[mid]<target)low=mid+1;
            else high=mid-1;
        }
        return last;
    }
    public int[] searchRange(int[] nums, int target) {

        return new int[]{firstocc(nums,target),lastocc(nums,target)};
    }
}